### PR TITLE
add micro-survey-banner with nps and pmf questions

### PR DESCRIPTION
**Pull request by groovecoder**
Adds a new banner along the top of the site for micro-surveys to track our NPS and PMF scores. The logic is essentially:

* If we're running a survey *recruitment* banner, that takes precedence over everything
* Else, flip a digital 50/50 coin and show either the NPS question or the PMF question
* If a user has answered either micro-survey in the last 30 days, don't show the survey again

### CODE REVIEW

**Comment by lesleyjanenorton** [comment]
Let's add a media query to hide this on narrower windows/smaller devices.

**Comment by groovecoder** [comment]
Added media query.

**Review by maxxcrawford — CHANGES_REQUESTED** [review]
I have some design notes but can help make those changes if you'd like! 

The bigger issue is that uBlock Origin is blocking `/js/analytics.js` from loading, which is obscuring part of the survey. 

I recommend using CSS to hide the survey (for those who are selected for the survey) and unhide it via JS in the `/js/analytics.js` file.

File: static/js/analytics.js [hunk 1]
```diff
@@ -117,6 +117,106 @@ function openOauth(clickEvt, entrypointElem) {
...
+        li.dataset.eventAction = "submitted";
+        li.dataset.eventLabel = option;
+        li.dataset.ga = "send-ga-funnel-pings";
+        li.addEventListener("click", setSurveyedCookie);
+        surveyOptions.appendChild(li);
+      });
+      break;
+    }
+  }
+  resetBodyPadding();
```
Comment (maxxcrawford) [inline static/js/analytics.js:217]: I recommend adding a function here to unhide the survey for selected users.

File: static/css/style.css [hunk 2]
```diff
@@ -431,24 +431,38 @@ header {
   transition: all 0.5s ease;
 }
 
-.recruitment-banner {
+.recruitment-banner, .micro-survey-banner {
```
Comment (maxxcrawford) [inline static/css/style.css:434]: Hide the survey by default with CSS:

```suggestion
.recruitment-banner, .micro-survey-banner {
  display: none;
```

**Review by maxxcrawford — CHANGES_REQUESTED** [review]

**Comment by maxxcrawford** [comment]
Revised style of the survey bar: 
- Increased background contrast
- Made clickable elements look more like buttons (padding, background color, rounded corners) 

PMF: 
image

NPS:
image

File: static/js/analytics.js [hunk 3]
```diff
@@ -117,6 +117,156 @@ function openOauth(clickEvt, entrypointElem) {
...
+          ga("send", "event",
+            eventData.eventCategory,
+            eventData.eventAction,
+            eventData.eventLabel,
+            eventData.eventValue,
+            {
+              dimension1: eventData.eventLabel,
+              metric1: 1,
+              metric2: eventData.eventValue,
+              metric3: eventData.npsValue,
```
Comment (groovecoder) [inline static/js/analytics.js:188]: Note: these metric indices are defined in GA:
image

File: static/js/analytics.js [hunk 4]
```diff
@@ -117,6 +117,156 @@ function openOauth(clickEvt, entrypointElem) {
...
+      let countMetric = "metric4";
+      let rankMetric = "metric5";
+      if (surveyType === "credibility") {
+        countMetric = "metric6";
+        rankMetric = "metric7";
+      }
+      if (surveyType === "appearance") {
+        countMetric = "metric8";
+        rankMetric = "metric9";
+      }
```
Comment (groovecoder) [inline static/js/analytics.js:231]: Note: these metric indices are defined in GA:
image

**Review by maxxcrawford — APPROVED** [review]
LGTM!  🚀 

I can't speak to the analytics, so let's be sure to confirm live data is capturing/reporting as expected once deployed.

**Pull request merged**